tests/unit/s2n_x509_validator_test.c

```diff
@@ -1262,6 +1381,116 @@ int main(int argc, char **argv)
+     * Test OCSP validation at various offsets from update times.
+     * libcrypto ASN1 comparison calculates differences in terms of days and seconds,
+     * so try mocking the system time to a collection of more than day & less
+     * than day differences.
+     * The T's in the below diagram represent test cases that should fail
+     * The F's represent test cases that should succeed
+     *    S2N_ERR_CERT_INVALID                            S2N_ERR_CERT_EXPIRED
+     *          |   |                                        |   |
+     *          v   v                                        v   v
+     *          F   F   T   T                        T   T   F   F
+     *          v   v   v   v                        v   v   v   v
+     * <----------|---|---|----------------------------|---|---|--->
+     *                ^                                    ^
+     *       this update                                 next update
+     *                |---|
+     *                  one day
+     *
+     * If this test is failing make sure that the this_update_timestamp_nanoseconds
+     * matches the actual timestamp of ocsp_response_early_expire.der
+     *
+     * openssl ocsp -respin ocsp_response_early_expire.der -text -noverify | grep "This Update"
+     */
+    {
+        /* Apr 28 22:11:56 2023 GMT */
+        uint64_t this_update_timestamp_nanoseconds = (uint64_t) 1682719916 * ONE_SEC_IN_NANOS;
+
+        /* Apr 28 22:11:56 2023 GMT */
+        uint64_t next_update_timestamp_nanoseconds = (uint64_t) 2082838316 * ONE_SEC_IN_NANOS;
+
+        uint64_t one_hour_nanoseconds = (uint64_t) 60 * 60 * ONE_SEC_IN_NANOS;
+        uint64_t one_day_nanoseconds = 24 * one_hour_nanoseconds;
+
+        uint64_t times[] = {
+            this_update_timestamp_nanoseconds - (one_day_nanoseconds + one_hour_nanoseconds),
+            this_update_timestamp_nanoseconds - one_hour_nanoseconds,
+            this_update_timestamp_nanoseconds + one_hour_nanoseconds,
+            this_update_timestamp_nanoseconds + (one_day_nanoseconds + one_hour_nanoseconds),
+            next_update_timestamp_nanoseconds - (one_day_nanoseconds + one_hour_nanoseconds),
+            next_update_timestamp_nanoseconds - one_hour_nanoseconds,
+            next_update_timestamp_nanoseconds + one_hour_nanoseconds,
+            next_update_timestamp_nanoseconds + (one_day_nanoseconds + one_hour_nanoseconds)
+        };
+
+        int expected_result[] = {
+            S2N_ERR_CERT_INVALID,
+            S2N_ERR_CERT_INVALID,
+            S2N_ERR_OK,
+            S2N_ERR_OK,
+            S2N_ERR_OK,
+            S2N_ERR_OK,
+            S2N_ERR_CERT_EXPIRED,
+            S2N_ERR_CERT_EXPIRED,
+        };
+
+        EXPECT_EQUAL(s2n_array_len(times), s2n_array_len(expected_result));
+
+        for (int i = 0; i < s2n_array_len(times); i++) {
+            /* print out test case for easier debugging from logs in CI */
+            fprintf(stdout, "\n Running Test Case  %i\n", i);
```

Comment:
I don't have a particularly strong opinion on this, but I do think it could be useful. The reason I added it is because I was trying to debug the failure that I encountered in an earlier CI run, but the only information was that the cert was untrusted and line that it failed (which was in the loop), so I didn't know which test case actually failed.

We already have stdout printing in 
- s2n_mem_usage_test.c
- s2n_server_new_session_ticket_test.c
- s2n_alerts_test.c
Although to be fair it doesn't look like those print as frequently as this would.

Additionally, CTest swallows all test output, so it wouldn't be printed unless the test actually fails.

Reply:
Meh, I think you've convinced me on this. To be fair my approach was to reproduce the failure locally, at which point why both including the print statement for CI? Maybe something to be revisited in the future if it causes a lot of pain.